### PR TITLE
Handle NVFP4 MoE w1/w3 weight_scale_2 mismatch by rescaling block scales

### PR DESCRIPTION
**Pull request by yhyang201**
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

**Review by ispobock — APPROVED** [review]
Need to test accuracy and make sure other nvfp4 models are not affected.

File: python/sglang/srt/layers/quantization/modelopt_quant.py [hunk 1]
```diff
@@ -1479,15 +1477,34 @@ def process_weights_after_loading(self, layer: torch.nn.Module) -> None:
...
-                    "Accuracy may be affected."
+                    "w1_weight_scale_2 does not match w3_weight_scale_2. "
+                    "Taking the max and rescaling block scales to compensate."
                 )
-
-            w13_weight_scale_2 = layer.w13_weight_scale_2[:, 0]
+                # Take the max of w1 and w3 global scales per expert,
+                # then adjust block scales so the product (block_scale *
+                # global_scale) remains unchanged.
+                max_scale = torch.max(w1_scale, w3_scale)  # [num_experts]
```
Comment (Edwardf0t1) [inline python/sglang/srt/layers/quantization/modelopt_quant.py:1491]: Does this improve accuracy? I'm curious if we have run any benchmarks on any models with the change.

Comment (yhyang201) [reply python/sglang/srt/layers/quantization/modelopt_quant.py:1491]: I tried it, but it didn’t improve the accuracy, so I changed it to draft.
Fortunately, most models didn’t encounter this issue — only the third layer of Kimi K2.5 has this problem. I’m not sure why either.
